### PR TITLE
add i18n

### PR DESCRIPTION
**Pull request by markkaylor**

### CODE REVIEW

File: starter/components/locale-switch.js
Comment (markkaylor) [inline starter/components/locale-switch.js]: this is throwing an error right now...will fix

**Comment by markkaylor** [comment]
Hey @remidej can you take a look?

- I'm using the locale code as the slug if it is not the defaultLocale
- If the next cookie is set to something other than the defaultLocale, for example `default = en, cookie = fr`, then `/` will redirect to `/fr`
- However, /[localeCode] throws a warning.  For some reason the `pageProps` are null on first load in _app.  Not sure why since they are not null when loggingl in `getInitialProps`

File: starter/pages/api/preview.js
Comment (markkaylor) [inline starter/pages/api/preview.js:6]: add this back

**Comment by remidej** [comment]
Hey @markkaylor, sorry for checking this so late. The new locale switch dropdown is beautiful ✨ 

I can't seem to get the warning you're talking about. Here's what happens on my end:

- The `/en` routes work well, they behave like before, just with the added prefix
- The root URL redirects to `/fr`, which is a 404, but it has a proper french navbar
- Navigating between the actual french URLs works well

**Comment by markkaylor** [comment]
No worries @remidej kudos to @rgesulfo for the dropdown, I dig it too.  

Ok yeah for me it has a 404 but then redirects to the proper fr homepage, maybe we can take a look together tomorrow.

File: starter/components/locale-switch.js
Comment (markkaylor) [inline starter/components/locale-switch.js]: works in develop, but with a production build the `initialLocale` from the cookie set in `_app` doesn't seem to be updating...

File: starter/pages/_app.js
Comment (markkaylor) [inline starter/pages/_app.js]: When accessing `/fr` on first load, `pageProps` is null, hits the 404 then redirects to the correct `/fr`.  `pageProps` is not null in `getInitialProps`.  This only happens in development, with a production build everything works as expected.

File: starter/components/locale-switch.js
Comment (markkaylor) [reply starter/components/locale-switch.js]: This was the problem
```
locale !== initialLocale
```

**Comment by markkaylor** [comment]
@remidej @alexandrebodin - Updated to use a string as a slug and the same slug for both locales. Thanks for your help :)

If you have time to take a look that would be awesome.  Here's Gatsby as well: https://github.com/strapi/strapi-starter-gatsby-corporate/pull/14

**Review by remidej — APPROVED** [review]
I'm happy to say that this looks great!

**Comment by remidej** [comment]
But let's wait for the template repo to be updated first before we merge this

**Pull request merged**
